platform/ios/app-swift/Sources/MultipleImagesExample.swift

```diff
@@ -0,0 +1,72 @@
+
+import MapLibre
+import SwiftUI
+import UIKit
+
+// #-example-code(MultipleImagesExample)
+class MultipleImagesExample: UIViewController, MLNMapViewDelegate {
+    var mapView: MLNMapView!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        mapView = MLNMapView(frame: view.bounds, styleURL: AMERICANA_STYLE)
+        mapView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        mapView.tintColor = .darkGray
+
+        let glacierPoint = CLLocationCoordinate2D(latitude: 37.72836462778902, longitude: -119.57352616838511)
+        mapView.setCenter(glacierPoint, animated: false)
+        mapView.zoomLevel = 12
+
+        mapView.delegate = self
+        view.addSubview(mapView)
+    }
+
+    // Wait until the style is loaded before modifying the map style.
+    func mapView(_: MLNMapView, didFinishLoading style: MLNStyle) {
+        let source = MLNVectorTileSource(identifier: "pois-nps", configurationURL: URL(string: "mbtiles://\(Bundle.main.bundlePath)/pois-nps.mbtiles")!)
+
+        style.addSource(source)
+
+        let imagesToAdd = [
+            ["nps-restrooms", "restrooms"],
+            ["nps-trailhead", "trailhead"],
+            ["nps-viewpoint", "viewpoint"],
+        ]
+
+        for imageInfo in imagesToAdd {
+            if let imageName = imageInfo.first, let imageKey = imageInfo.last {
+                if let image = UIImage(named: imageName) {
+                    style.setImage(image, forName: imageKey)
+                } else {
+                    print("Failed to load \(imageName)")
+                    return
+                }
+            }
+        }
+
+        let imageLayer = MLNSymbolStyleLayer(identifier: "npc-poi-images", source: source)
+        imageLayer.sourceLayerIdentifier = "pois"
+        imageLayer.iconImageName = NSExpression(mglJSONObject: [
+            "match", ["get", "POITYPE"],
+            "Restroom", "restrooms",
+            "Trailhead", "trailhead",
+            "Viewpoint", "viewpoint",
+            "",
```

Comment:
Does this work without doing the `["image", "restrooms"]`?

Related to: https://github.com/maplibre/maplibre-native/issues/331#issuecomment-2220468477

Documentation make is sound that we need to use the `["image", "image-name"]`. [Docs](https://maplibre.org/maplibre-style-spec/expressions/#image)

I'm confused! 🤔

Reply:
That would be true if there was a property `iconImage`, however in iOS you set `iconImageName` directly.

Reply:
And yes it works...

Reply:
Both work actually.